File: FPG.Core/Algorithms/SaltGeneration/WPSaltGenerationAlgorithm.cs
using System;
using System.Text;

using FPG.Interfaces;

namespace FPG.Algorithms
{
    public class WPSaltGenerationAlgorithm : ISaltGenerationAlgorithm
    {
        public Int32 Length { get; set; } = 64;

        public string Generate()
        {
            StringBuilder sb = new StringBuilder();

            for (Int32 i = 0; i < this.Length; i++)
            {
                sb.Append(WPSaltGenerationAlgorithm.Characters.Substring
                    (this.Random.Next(0, WPSaltGenerationAlgorithm.Characters.Length - 1), 1));
            }

            return sb.ToString();
        }

        private Random Random => new Random();
        private const String Characters
            = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_ []{}<>~`+=,.;:/?|";
    }
}

File: lib/core/algorithms/saltgeneration/WPSaltGenerationAlgorithm.dart
import 'dart:math';

import 'package:fpg_mobile/core/interfaces/ISaltGenerationAlgorithm.dart';

class WPSaltGenerationAlgorithm implements ISaltGenerationAlgorithm {
  int length = 64;

  @override
  String generate() {
    Random random = Random();
    StringBuffer sb = StringBuffer();

    for (int i = 0; i < length; i++) {
      int r = random.nextInt(_characters.length - 1);
      sb.write(_characters.substring(r, r + 1));
    }

    return sb.toString();
  }

  static const String _characters =
      r"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_ []{}<>~`+=,.;:/?|";
}
